Python/marshal.c

```diff
@@ -1998,12 +2065,15 @@ _PyCode_Hydrate(PyCodeObject *code)
     rf.end = s + n;
     rf.depth = 0;
     rf.refs = ctx->refs;
+    Py_XINCREF(rf.refs);
     rf.refs_pos = code->co_hydra_refs_pos;
     rf.ctx = ctx;
     ctx->code = code;
 
     PyObject *result = read_object(&rf);
-
+    Py_XDECREF(rf.refs);
+    Py_XDECREF(code->co_hydra_context);
```

Comment:
To be sure, this is only ever non-NULL if read_object() encountered an error, right? We're basically making it so that you can hydrate a code object only once. (That seems reasonable to me.)

Reply:
> To be sure, this is only ever non-NULL if read_object() encountered an error, right?

If there is no error it gets set to NULL? I don't see that.

> We're basically making it so that you can hydrate a code object only once. (That seems reasonable to me.)

That's the idea, to get rid of the context once we're done with it.